### PR TITLE
Dockerisation: Added dockerfile and updated Readme

### PR DESCRIPTION
**Pull request by Cawllec**
Adds a basic dockerfile to the Laravel 5.6 example, allow it to be run in a local container.

This not only will allow us to get the examples running inside a container faster, but also is a stepping stone toward end-to-end/maze tests.

An updated readme for the example has also been provided.

## Tests

The docker container was manually run, and confirmed to be delivering notifications to Bugsnag.

## Review

The example needs to be tested so that the example can be built and run out of the box.

### CODE REVIEW

File: example/laravel56/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+FROM php:latest
+
+RUN apt-get update && \
+  apt-get install -y --no-install-recommends git zip unzip wget
+
+RUN mkdir /app
```
Comment (martin308) [inline example/laravel56/Dockerfile]: using the `WORKDIR` command below will create the required directory

File: example/laravel56/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+WORKDIR /app
+
+COPY . /app/
```
Comment (martin308) [inline example/laravel56/Dockerfile]: everything is relative from the `WORKDIR` once it is set so we can use relative paths here

File: example/laravel56/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+RUN /app/install_composer.sh
```
Comment (martin308) [inline example/laravel56/Dockerfile]: same here with a relative path

File: example/laravel56/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+RUN cd /app/ && \
```
Comment (martin308) [inline example/laravel56/Dockerfile]: you shouldn't need to change directory here either due to using `WORKDIR`

File: example/laravel56/install_composer.sh [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+then
+    >&2 echo 'ERROR: Invalid installer signature'
+    rm composer-setup.php
+    exit 1
+fi
+
+php composer-setup.php
+RESULT=$?
+rm composer-setup.php
+exit $RESULT
```
Comment (martin308) [inline example/laravel56/install_composer.sh]: would it be better to just use the composer image directly so we don't need this script?

File: example/laravel56/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,16 @@
+FROM php:latest
+
+RUN apt-get update && \
+  apt-get install -y --no-install-recommends git zip unzip wget
```
Comment (martin308) [inline example/laravel56/Dockerfile]: it'll make it easier to maintain this dockerfile if we organize these like this

```
RUN apt-get update && \
  apt-get install -y --no-install-recommends \
  git \
  unzip \
  wget \
  zip
```

File: example/laravel56/install_composer.sh (hunk 5 above)
Comment (Cawllec) [reply example/laravel56/install_composer.sh]: The composer image only runs composer, and isn't designed for running PHP servers unfortunately.

**Comment by Cawllec** [comment]
Thanks for the feedback Martin. I've made some updates based on your (and @mikeewhite's in person) feedback, if you could give it another quick look over, that would be great.

**Comment by snmaynard** [comment]
You cant run the official composer image. It is just a wrapper for the cli. https://github.com/composer/docker/issues/44#issuecomment-365562680 as an example. You might be able to use it as a base and then go from there, but I think this is better.

**Pull request merged**